src/Hyperion/Run.hs

```diff
@@ -81,6 +89,20 @@ runBenchmark samplef bk0 =
       put xs
       return x
 
+-- | Run a benchmark on varying sample sizes
+runBenchmarkVaryingSizes :: [(Batch () -> IO Sample)] -> Benchmark -> IO (HashMap Text [Sample])
+runBenchmarkVaryingSizes samplef bk0 = formatSamples $ sequence (map (\samplef' -> runBenchmark samplef' bk0) samplef)
+
+-- | Set of functions to massage the data into the appropriate type
+formatSamples :: IO([HashMap Text Sample]) -> IO (HashMap Text [Sample])
+formatSamples samples = concatenateSamples $ expandSamples samples
+
+expandSamples :: IO ([HashMap Text Sample]) -> IO ([HashMap Text [Sample]])
+expandSamples samples = fmap (map (fmap (\x -> [x]))) samples
```

Comment:
[style] `\x -> [x]` is same as `return` (but not sure if it more helpful for reader), alternatively it can be written as happy robot: `(:[])`. I think everybody will understand happy robot (possibly expect first time in their life) and then when they see it in code their mood may improve a bit.